### PR TITLE
Add Thai translations

### PR DESCRIPTION
**Pull request by taperec**
I follow this guide (https://github.com/commaai/openpilot/tree/master/selfdrive/ui/translations) to add Thai language translations.

I already run test command.

This pull request is very latest.
Fix spelling mistakes and improve sentences from this pull request. (https://github.com/commaai/openpilot/pull/25115)

Thank you.

### CODE REVIEW

File: selfdrive/ui/translations/main_th.ts [hunk 1]
```diff
@@ -0,0 +1,1274 @@
...
+        <source>Updating files: </source>
+        <translation>กำลังอัปเดตไฟล์: </translation>
+    </message>
+</context>
+<context>
+    <name>MapETA</name>
+    <message>
+        <location filename="../qt/maps/map.cc" line="622"/>
+        <source>eta</source>
+        <translation>เวลาถึงปลายทางโดยประมาณ</translation>
```
Comment (sshane) [inline selfdrive/ui/translations/main_th.ts:343]: Unfortunately this needs to be quite short to fit in the provided space with the current design. Shall we use `eta` instead?

**Comment by sshane** [comment]
Mostly looks good though! Most translations are as long or shorter than the English text, so shouldn't have any issues with design

File: selfdrive/ui/translations/main_th.ts [hunk 2]
```diff
@@ -0,0 +1,1327 @@
...
+    <message>
+        <location filename="../qt/util.cc" line="21"/>
+        <source>openpilot</source>
+        <translation>openpilot</translation>
+    </message>
+    <message numerus="yes">
+        <location filename="../qt/util.cc" line="82"/>
+        <source>%n minute(s) ago</source>
+        <translation type="unfinished">
+            <numerusform></numerusform>
```
Comment (sshane) [inline selfdrive/ui/translations/main_th.ts:648]: Could you add the few unfinished translations with the rebase?

File: selfdrive/ui/translations/main_th.ts (hunk 1 above)
Comment (taperec) [reply selfdrive/ui/translations/main_th.ts:349]: Yes, we can use "eta" instead. No Thai word to replace this. I will update it.

File: selfdrive/ui/translations/main_th.ts (hunk 2 above)
Comment (taperec) [reply selfdrive/ui/translations/main_th.ts:648]: Yes, I will do it.

Comment (sshane) [reply selfdrive/ui/translations/main_th.ts:648]: Hello. Any updates on doing these?

Comment (taperec) [reply selfdrive/ui/translations/main_th.ts:648]: > 

Sorry, I will try to finish it today.

Comment (sshane) [reply selfdrive/ui/translations/main_th.ts]: thanks!

File: selfdrive/ui/translations/main_th.ts [hunk 3]
```diff
@@ -0,0 +1,1303 @@
...
+        <source>Wrong password</source>
+        <translation>รหัสผ่านผิด</translation>
+    </message>
+</context>
+<context>
+    <name>NvgWindow</name>
+    <message>
+        <location filename="../qt/onroad.cc" line="218"/>
+        <source>km/h</source>
+        <translation>ก.ม./ช.ม.</translation>
```
Comment (sshane) [inline selfdrive/ui/translations/main_th.ts:495]: Not sure, is `km/h` universally recognizable, or is the translation more recognizable?

File: selfdrive/ui/translations/main_th.ts [hunk 4]
```diff
@@ -0,0 +1,1303 @@
...
+        <source>Wrong password</source>
+        <translation>รหัสผ่านผิด</translation>
+    </message>
+</context>
+<context>
+    <name>NvgWindow</name>
+    <message>
+        <location filename="../qt/onroad.cc" line="218"/>
+        <source>km/h</source>
+        <translation>กม./ชม</translation>
```
Comment (sshane) [inline selfdrive/ui/translations/main_th.ts:495]: Do these shorter translations make sense? @taperec

File: selfdrive/ui/translations/main_th.ts [hunk 5]
```diff
@@ -0,0 +1,1303 @@
...
+    </message>
+    <message>
+        <location filename="../qt/maps/map.cc" line="631"/>
+        <source>min</source>
+        <translation>นาที</translation>
+    </message>
+    <message>
+        <location filename="../qt/maps/map.cc" line="635"/>
+        <source>hr</source>
+        <translation>ช.ม.</translation>
```
Comment (sshane) [inline selfdrive/ui/translations/main_th.ts:355]: Google Translate says this means "watch," does this mean "hour"?

**Comment by sshane** [comment]
Hmm, on PC this looks fine, but the typeface on the comma three is small and pixelated. Moving to draft for now

File: selfdrive/ui/translations/main_th.ts (hunk 5 above)
Comment (taperec) [reply selfdrive/ui/translations/main_th.ts:355]: watch in Thai is ชม (no dot)

hour and hours in Thai is ชั่วโมง (no plural)
ช.ม. (with dot between letter) is short for ชั่วโมง, like hr. in english.

File: selfdrive/ui/translations/main_th.ts (hunk 4 above)
Comment (taperec) [reply selfdrive/ui/translations/main_th.ts:495]: short from of Thai word must have dot between letter but it's ok with single end dot or no dot.

ก.ม./ช.ม. <-- This is formal.
กม./ชม. <-- also can use this.
กม/ชม <-- also can use this.

Comment (taperec) [reply selfdrive/ui/translations/main_th.ts]: Already update short form of words.

**Comment by adeebshihadeh** [comment]
> Hmm, on PC this looks fine, but the typeface on the comma three is small and pixelated. Moving to draft for now. @adeebshihadeh want to fix the AGNOS fonts for this?

Yes, @sshane can you make a PR to add the Thai Inter font to agnos-builder? I'll make sure it goes in the next agnos release. For now, think we should merge this without being able to select it in the UI.

**Comment by sshane** [comment]
I don't think Inter supports Thai https://rsms.me/inter/

**Comment by sshane** [comment]
Thanks for the PR! The fonts aren't yet in AGNOS, and there's an issue with line height in the prime widget, so merging this language in shadow mode for now.

PXL_20220823_232342231

**Pull request merged**